src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -133,6 +149,76 @@ public function validateDefinition($element, $objectDefinition = null, JsonPoint
     }
 
     /**
+     * Converts a value to boolean. For example, "true" becomes true.
+     * @param $value The value to convert to boolean
+     * @return bool|mixed
+     */
+    protected function toBoolean($value)
+    {
+        if($value === "true"){
+            return true;
+        }
+
+        if($value === "false"){
+            return false;
+        }
+
+        return $value;
+    }
+
+    /**
+     * Converts a numeric string to a number. For example, "4" becomes 4.
+     *
+     * @param mixed $value The value to convert to a number.
+     * @return int|float|mixed
+     */
+    protected function toNumber($value)
+    {
+        if(is_numeric($value)) {
+            return $value + 0; // cast to number
+        }
+
+        return $value;
+    }
+
+    protected function toInteger($value)
+    {
+        if(ctype_digit ($value)) {
+            return $value + 0; // cast to number
```

Comment:
I am referring to an explicit cast instead of an operation that results in an implicit cast:

```php
return (int) $value;
```

For the `toNumber` method it makes sense to use the operation to produce an integer or a float, but for a whole number simply casting to the desired type is more expressive, in my opinion.